src/util/README.md

```diff
@@ -12,17 +12,57 @@ CPROVER codebase.
 
 \subsection irept_section irept: a 4-tuple (data, named-sub, comments, sub)
 
-See documentation at \ref irept.
+See detailed documentation at \ref irept.
 
-As that documentation says, [irept](\ref irept)s are generic tree nodes. You
-should think of them as having a single string ([data](irept::data), actually
-an \ref irep_idt) and lots of child nodes, some of which are numbered
+[irept](\ref irept)s are generic tree nodes. You
+should think of each node as holding a single string ([data](irept::data),
+actually an \ref irep_idt) and lots of child nodes, some of which are numbered
 ([sub](\ref irept::dt::sub)) and some of which are labelled, and the label
 can either start with a “\#” ([comments](\ref irept::dt::comments)) or without
 one ([named_sub](\ref irept::dt::named_sub)). The meaning of the “\#” is that
 this child should not be considered important, for example it shouldn't be
```

Comment:
While at it: "should not be considered important" should be rephrased. It's about semantic equivalence of programs, with two programs (or expressions within programs) being semantically (but not necessarily syntactically) equivalent even if entries with a `#` prefix are different.